src/plugins/intel_gna/transformations/transpose_to_pwl.hpp

```diff
@@ -0,0 +1,333 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <cmath>
+#include <stdexcept>
+
+#include <transformations_visibility.hpp>
+
+#include <ngraph/ngraph.hpp>
+#include <ngraph/pass/graph_rewrite.hpp>
+#include "ngraph/pattern/matcher.hpp"
+#include <ngraph/opsets/opset8.hpp>
+#include <legacy/ngraph_ops/power.hpp>
+
+namespace GNAPluginNS {
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief TransposeToPwl transformation replaces suitable activation function with pwl
+ */
+class TRANSFORMATIONS_API TransposeToPwl : public ngraph::pass::MatcherPass {
+public:
+    NGRAPH_RTTI_DECLARATION;
+    TransposeToPwl(double max_error_percent);
+};
+
+namespace details {
+struct Pwl {
+    double m;
+    double b;
+    double alpha;
+}; // struct Pwl
+
+template<typename T>
+struct Function;
+
+template<>
+struct Function<ngraph::opset8::Sigmoid> {
+    static const char* name() {
+        return "Sigmoid";
+    }
+
+    double get_value(double x) const {
+        return 0.5 * (1.0 + tanh(x / 2.0));
+    }
+
+    double first_derivative(double x) const {
+        return get_value(x) * (1.0 - get_value(x));
+    }
+
+    static double lower_bound() {
```

Comment:
Does it mean that with new approach pwl segments will be calculated in the range [-SigmoidDomain, SigmoidDomain] like it was done in legacy? not for the range specified in FQ statistics? In this case we should be sure that the segments  which can be used in the model have ends which are not overflown during quantization.